lib/datadog/tracing/contrib/extensions.rb

```diff
@@ -63,6 +63,10 @@ def configure(&block)
             # Reconfigure core settings
             super(&block)
 
+            # Configuration should be considered RO past this point:
+            # - Patcher's `patch` must not modify configuration.
+            # - use Integration's `subconfigure` instead
+
```

Comment:
I was wondering about calling `subconfigure` here instead:

- gives access to full config, not just tracing's, which would allow cross-product integration subconfiguration (e.g appsec could enable tracing)
- config is final state after whole `configure`'s `&block` is executed vs at the time the `instrument :sinatra` line is